mmv1/products/compute/api.yaml

```diff
@@ -12894,6 +12895,24 @@ objects:
           - next_hop_ip
           - next_hop_vpn_tunnel
           - next_hop_ilb
+      - !ruby/object:Api::Type::String
```

Comment:
You can remove this section altogether

Reply:
So the reason to have different field configurations for ga and beta is twofold:

1. The current field definition is a self link to a forwarding rule. If I don't change the field to a string the example fails with an error like this 

```
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccComputeRoute_routeIlbVip -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccComputeRoute_routeIlbVipExample
=== PAUSE TestAccComputeRoute_routeIlbVipExample
=== CONT  TestAccComputeRoute_routeIlbVipExample
    provider_test.go:278: Step 1/2 error: Error running apply: exit status 1

        Error: Error waiting to create Route: Error waiting for Creating Route: The resource 'projects/xxx/regions/us-central1/forwardingRules/10.0.1.2' was not found


          with google_compute_route.route-ilb,
          on terraform_plugin_test.tf line 74, in resource "google_compute_route" "route-ilb":
          74: resource "google_compute_route" "route-ilb" {

--- FAIL: TestAccComputeRoute_routeIlbVipExample (154.36s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	154.386s
FAIL
```
That's why I came up with the idea of having a different diff suppress function. Is there any way to allow the field to be of ResourceRef type but still allow an IP to be passed in?

2. The other reason is that the option to pass an IP is still beta, so that option shouldn't be available in the ga provider. If I just make the change in the existing field, the ga provider would be able to pass an IP to the field. Not a big deal but breaks the pattern of having only GA features in the GA provider.

Reply:
Let's unify the change. Can we remove the other section and just retain this one. We can have the description alert the user that ip will not work in beta.

Reply:
I've removed the second field (the one using ResourceRef) and updated the description.

Note that right now the GA provider does allow an IP. How can we ensure it doesn't?

Reply:
We can't ensure it doesn't but we will just let the api fail in this case. The user can read the documentation to understand why.